### DISCUSSION TITLE
The back button does not work in LibreTube

### DISCUSSION THREAD

**Discussion by Pamilg8** [libre-tube/LibreTube#5716] [General]
### Steps to reproduce

1. Enter LibreTube
2. Press the back button on your device
3. Back button doesn't work. It does not work in any section

### Expected behavior

The back button should work without problems 

### Actual behavior

Back button functionality is not working in the app. 
This problem happened suddenly 

### LibreTube version

0.21.1 Debug

### Android version

Android 12 vs 13 ( One UI Core 4.1, One UI 5.1)

### Other details

https://github.com/libre-tube/LibreTube/assets/149590337/b98c6b98-d005-46b8-ae1a-0d3f93f0ce60

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by Bnyro** [comment]
Good catch, it seems like some dependency update now forces us to explicitly enable handling back presses - probably due to the new back gesture invocation APIs added in Android 14.

**Comment by Pamilg8** [comment]
> Good catch, it seems like some dependency update now forces us to explicitly enable handling back presses - probably due to the new back gesture invocation APIs added in Android 14.

I tried the debug version and it didn't fix the problem

**Comment by Bnyro** [comment]
Strange, it's fixed for me.

**Comment by Pamilg8** [comment]
> Strange, it's fixed for me.

Can you please keep the issue open? Maybe someone helped me😔

**Comment by t-e-s-tweb** [comment]
> Strange, it's fixed for me.

tried the latest commits but it isn't fixed. It works though by reverting this commit https://github.com/libre-tube/LibreTube/commit/ce8ed66e927ab2232eb29f756692e80523b34ea3

**Comment by Bnyro** [comment]
> > Strange, it's fixed for me.
> 
> tried the latest commits but it isn't fixed. It works though by reverting this commit [ce8ed66](https://github.com/libre-tube/LibreTube/commit/ce8ed66e927ab2232eb29f756692e80523b34ea3)
>
Reverting this commit only or using a build before that commit (so also reverting any other changes)?

**Comment by t-e-s-tweb** [comment]
> > > Strange, it's fixed for me.
> > 
> > 
> > tried the latest commits but it isn't fixed. It works though by reverting this commit [ce8ed66](https://github.com/libre-tube/LibreTube/commit/ce8ed66e927ab2232eb29f756692e80523b34ea3)
> 
> Reverting this commit only or using a build before that commit (so also reverting any other changes)?

only this commit. 

i use this command in GitHub workflow 

- name: Revert Commit
        run: |
          git revert -v -n -m 1 ce8ed66 --no-edit -X theirs

**Comment by t-e-s-tweb** [comment]
which device are you the testing it on? this and that full screen bug aren't reproducible. I am on MiUi. i wonder what the author of this issue is running as OS.

**Comment by Pamilg8** [comment]
I test LibreTube on different devices.( Android 12, One UI Core 4.1)
Or Android 13, One UI 5.1
Many Samsung devices have the problem
The back button works when the device is offline

**Comment by akano12** [comment]
Can't reproduce either.
Samsung S20 A13, can you try latest debug?

**Comment by Pamilg8** [comment]
> Can't reproduce either. Samsung S20 A13, can you try latest debug?

I have tried the latest debug on samsung a32 and samsung a21s, a31 and the problem is not fixed. Samsung a13 runs full One UI. You try A21s and a12 running One UI Core version. A32 and a31 are running full One UI 5.1 and One UI 4.1 but they have the same problem. The S20 is an expensive phone. Most users use a budget phone

**Comment by Pamilg8** [comment]
1709460796.079 10696 21087 21087 I ViewRootImpl@f363c38[Default]: ViewPostIme key 0
1709460796.166 10696 21087 21087 I ViewRootImpl@f363c38[Default]: ViewPostIme key 1
1709460800.235 10696 21087 21087 I ViewRootImpl@f363c38[Default]: MSG_WINDOW_FOCUS_CHANGED 0 1
1709460800.264 10696 21087 21087 D InputTransport: Input channel destroyed: 'ClientS', fd=140

However, when I go back to the home page from the search section, the button works. In other cases, it never works

**Comment by FineFindus** [comment]
I can reproduce the problem on the latest master and can confirm that reverting to 0b059ccc12ab5a6df89c3c61f34680ad18db5471 fixes it. Interestingly, it seemed to work on my emulator, so maybe it only affects lower versions of Android?

**Reply by t-e-s-tweb** [reply]
finally someone was able to reproduce it. its present on latest debug too. but I wouldn't call android 12 a lower version android because according to statecounter as of Feb 2024 android 13 is only on ~28% of devices, rest are on 12 and below. 
also on the latest commits reverting is causing builds to fail. i hope a solution of found soon before the code changes too much.
